src/handlers/serenity.rs

```diff
@@ -116,63 +181,78 @@ impl EventHandler for SerenityHandler {
                         .name("stop")
                         .description("Stops the bot and clears the queue")
+                        .default_permission(false)
                 })
                 .create_application_command(|command| {
                     command
                         .name("summon")
                         .description("Summons the bot in your voice channel")
+                        .default_permission(true)
                 })
                 .create_application_command(|command| {
                     command
                         .name("version")
                         .description("Displays the current version")
+                        .default_permission(true)
                 })
         })
         .await
-        .unwrap();
+        .unwrap()
     }
 
-    async fn interaction_create(&self, ctx: Context, interaction: Interaction) {
-        if let Interaction::ApplicationCommand(mut command) = interaction {
-            match command.data.name.as_str() {
-                "autopause" => autopause(&ctx, &mut command).await,
-                "clear" => clear(&ctx, &mut command).await,
-                "leave" => leave(&ctx, &mut command).await,
-                "np" => now_playing(&ctx, &mut command).await,
-                "pause" => pause(&ctx, &mut command).await,
-                "play" => play(&ctx, &mut command).await,
-                "playtop" => playtop(&ctx, &mut command).await,
-                "queue" => queue(&ctx, &mut command).await,
-                "remove" => remove(&ctx, &mut command).await,
-                "repeat" => repeat(&ctx, &mut command).await,
-                "resume" => resume(&ctx, &mut command).await,
-                "seek" => seek(&ctx, &mut command).await,
-                "shuffle" => shuffle(&ctx, &mut command).await,
-                "skip" => skip(&ctx, &mut command).await,
-                "stop" => stop(&ctx, &mut command).await,
-                "summon" => summon(&ctx, &mut command, true).await,
-                "version" => version(&ctx, &mut command).await,
-                _ => unimplemented!(),
+    async fn ensure_role(
+        &self,
+        ctx: &Context,
+        guild: GuildId,
+        role_name: &str,
+    ) -> Result<Role, SerenityError> {
+        let roles = guild.roles(&ctx.http).await.unwrap();
+        let role = roles.iter().find(|(_, role)| role.name == role_name);
+        match role {
+            Some((_, role)) => Ok(role.to_owned()),
+            None => {
+                guild
+                    .create_role(&ctx.http, |r| r.name(role_name).mentionable(true))
+                    .await
             }
-            .unwrap();
         }
     }
 
-    async fn voice_state_update(
-        &self,
-        ctx: Context,
-        guild: Option<GuildId>,
-        _old: Option<VoiceState>,
-        new: VoiceState,
-    ) {
+    async fn run_command(&self, ctx: &Context, command: &mut ApplicationCommandInteraction) {
```

Comment:
This isn't really a `run_command` but a more like a `register_command_callback` I'd say.

Reply:
No, this actually runs the command.

https://github.com/aquelemiguel/parrot/blob/jc/47-dj-permission/src/handlers/serenity.rs#L233

This actually runs the `autopause` command